.github/workflows/update.yml

```
@@ -1,22 +1,51 @@
-name: "Test"
+name: "index-generate"
 on:
   workflow_dispatch:
   schedule:
     # every sunday at 2:51
     - cron:  '51 2 * * 0'
 jobs:
-  tests:
-    runs-on: ubuntu-latest
+  index:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            indexName: linux
+          - os: macos-latest
+            indexName: darwin
+    name: Build database
     steps:
     - uses: actions/checkout@v2
     - uses: cachix/install-nix-action@v13
       with:
         nix_path: nixpkgs=https://github.com/NixOS/nixpkgs/archive/nixpkgs-unstable.tar.gz
-    - name: Build database
-      run: nix-shell -p nix-index --command "nix-index --db . 2>&1 | grep -v '+ generating index'"
+    - name: run nix-index
+      run: nix-shell -p nix-index --command "nix-index --db ./${{ matrix.indexName }}-index 2>&1 | grep -v '+ generating index'"
+    - name: upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ matrix.indexName }}
+        path: ./${{ matrix.indexName }}-index
+
+  upload-index:
+    runs-on: ubuntu-latest
+    needs: index
+    steps:
+    - name: download linux artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: linux
+        path: linux-index
+    - name: download darwin artifact
+      uses: actions/download-artifact@v2
+      with:
+        name: darwin
+        path: darwin-index
+
     - name: Get current date
       id: date
-      run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+      run: echo "::set-output name=date::$(date +'%Y-%m-%d-%S')"
```

Do we need seconds here?

```suggestion
      run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
```

---

had it for testing